src/pages/VeSDL/OnChainVote.tsx

```diff
@@ -58,22 +63,21 @@ export default function OnChainVote({
     errorText: string
   }>({ hasError: false, errorText: " " })
 
-  const gaugeNames = useMemo(
-    () =>
-      Object.keys(gauges)
-        .filter((lpAddress) => !gauges[lpAddress]?.isKilled)
-        .reduce((acc, curr) => {
-          const gaugeName = gauges[curr]?.gaugeName
-          const gaugeAddress = gauges[curr]?.address
-          if (gaugeName && gaugeAddress)
-            acc?.push({
-              address: gaugeAddress,
-              gaugeName: gaugeName,
-            })
-          return acc
-        }, [] as GaugeName[]),
-    [gauges],
-  )
+  const gaugeNames = useMemo(() => {
+    const gaugeNameObj = {} as GaugeName
```

Comment:
instead of pushing values into an object can you just return the final result as a map? also what is the point of useMemo here?

Reply:
I made `gaugeName` to select `gauge Name` buy its address easily. We need to display vote status for each Gauge rather than vote history.
From event of gauge vote, we can get `user`, `time`, weight`, `gauge_addr`.

So, it would be good to make gaugeName as object rather than map to find the gauge name easily.
And I used `useMemo` to prevent the recalculate the `gaugeName` every time when state of onChain vote change.
We only need to calculate when `gauges` change.